### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by Fene-87**
I use transactions to update and delete records. I also use my knowledge of aggregate functions and GROUP BY to answer analytical questions. I implemented the following requirements:
- Add a column named species to animals table.
- Add more data to the animals table.
- Inside a transaction, update the animals table by setting the species column to 'digimon' for all animals that have a name ending in 'mon' and 'pokemon' for all animals that don't have species already set, the commit.
Sql2DigimonPokemon
- Inside a transaction delete all records in the animals table, then roll back the transaction and verify if all records still exist.
Sql2DeleteRollback
- Inside a transaction:
  Delete all animals born after Jan 1st, 2022 then create a savepoint.
Sql2Savepoint

  Update all animals' weight to be their weight multiplied by -1 the rollback to savepoint.
Sql2RollbackToSavepoint

  Update all animals' weights that are negative to be their weight multiplied by -1 then commit transaction.
Sql2CommitTransaction

- Write queries to answer the following questions:
  How many animals are there?
Sql2AnimalsCount

  How many animals have never tried to escape?
Sql2EscapeCount

  What is the average weight of animals?
Sql2AvgWeight

  Who escapes the most, neutered or not neutered animals?
Sql2MostEscaped

  What is the minimum and maximum weight of each type of animal?
Sql2MaxMinWeight

  What is the average number of escape attempts per animal type of those born between 1990 and 2000?
Sql2AvgEscape

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -4,4 +4,11 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
     VALUES ('Agumon', '2020-02-03', 0, true, 10.23),
            ('Gabumon', '2018-11-15', 2, true, 8),
            ('Pikachu', '2021-01-07', 1, false, 15.04),
-           ('Devimon', '2017-05-12', 5, true, 11);
+           ('Devimon', '2017-05-12', 5, true, 11),
+           ('Charmander', '2020-02-08', 0, false, 11),
+           ('Plantmon', '2021-11-15', 2, true, 5.7),
+           ('Squirtle', '1993-04-02', 3, false, 12.13),
+           ('Angemon', '2005-06-12', 1, true, 45),
+           ('Boarmon', '2005-06-07', 7, true, 20.4),
```
Comment (danmainah) [inline data.sql]: - [ ] @Fene-87 great work on your project, however, you have changed the weight from a negative figure to a positive, which will affect the queries, kindly change the values and also update the 3 queries dealing with weight.

File: queries.sql [hunk 2]
```diff
@@ -15,3 +15,52 @@ SELECT * FROM animals WHERE neutered = true;
...
+SELECT * FROM animals;
+
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+COMMIT;
```
Comment (danmainah) [inline queries.sql:36]: - [ ] @Fene-87 great work on your query, however, you forgot to add a query before and after the commit to verify the changes were made and persisted after the commit.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Fene-87 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: queries.sql (hunk 2 above)
Comment (Fene-87) [reply queries.sql:36]: Thanks @danmainah I have implemented the changes

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @Fene-87, 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well-implemented suggestions by the previous reviewer.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**